craft_archives/repo/apt_key_manager.py

```diff
@@ -250,7 +250,8 @@ def install_package_repository_key(
         # If the keyring exists but does not contain the key, remove it and
         # install a fresh one.
         keyring_path = get_keyring_path(key_id, base_path=self._keyrings_path)
-        keyring_path.unlink(missing_ok=True)
+        with contextlib.suppress(NotADirectoryError):
```

Comment:
What's your opinion on using `if keyring_path.is_file()` as the 'guard' instead?

Reply:
We can do that, but it'll likely need to be more than just that to handle a situation like that. `if keyring_path.is_file() or keyring_path.is_symlink() or keyring_path.is_socket()` or so.

What about guarding it with `if keyring_path.parent.is_dir()`? This will still prevent the `NotADirectoryError` I'm suppressing, but it'll catch any form of borked keyring path that exists for reasons unknown.